backend/app/services/knowledge/knowledge_service.py

```
@@ -1262,11 +1420,10 @@ def update_document_content(
             .first()
         )
         if kb and kb.namespace != "default":
-            if not check_group_permission(
-                db, user_id, kb.namespace, GroupRole.Maintainer
-            ):
+            # Owner/Maintainer/Developer can edit documents (edit+ permission)
+            if not _check_kb_permission(db, kb, user_id, "edit"):
                 raise ValueError(
-                    "Only Owner or Maintainer can edit documents in this knowledge base"
+                    "Only Owner, Maintainer, or Developer can edit documents in this knowledge base"
                 )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)